Sources/SwiftParser/Patterns.swift

```diff
@@ -217,9 +217,20 @@ extension Parser {
       while !self.at(.eof, .rightParen) && keepGoing && loopProgress.evaluate(currentToken) {
         // If the tuple element has a label, parse it.
         let labelAndColon = self.consume(if: .identifier, followedBy: .colon)
-        let (label, colon) = (labelAndColon?.0, labelAndColon?.1)
+        var (label, colon) = (labelAndColon?.0, labelAndColon?.1)
+
+        if label == nil, colon == nil, peek().rawTokenKind == .identifier, peek().tokenText.isStartingWithUppercase {
+          label = consumeAnyToken()
```

Comment:
I think a comment explaining why we check for `isStartingWithUppercase` would be good here. Something like 

```swift
/// If we have something like `x SomeType`, use the indication that `SomeType` starts with a capital letter (and is thus probably a type name) as an indication that the user forgot to write the colon instead of forgetting to write the comma to separate two elements.
```